doc/languages-frameworks/idris.section.md

```diff
@@ -1,39 +1,113 @@
-Idris packages
-==============
+# Idris packages
 
-This directory contains build rules for idris packages. In addition,
-it contains several functions to build and compose those packages.
-Everything is exposed to the user via the `idrisPackages` attribute.
+## Installing Idris
 
-callPackage
-------------
+The easiest way to get a working idris version is to install the `idris` attribute:
 
-This is like the normal nixpkgs callPackage function, specialized to
-idris packages.
+```
+$ # On NixOS
+$ nix-env -i nixos.idris
+$ # On non-NixOS
+$ nix-env -i nixpkgs.idris
+```
 
-builtins
----------
+This however only provides the `prelude` and `base` libraries. To install additional libraries:
 
-This is a list of all of the libraries that come packaged with Idris
-itself.
+```
+$ nix-env -iE 'pkgs: with pkgs.idrisPackages; with-packages [ contrib pruviloj ]'
```

Comment:
iirc for similar attributes we recommend the form `pkgs.idrisPackages.with-packages` not `with ...; with-packages` -- that seems more obvious to me here as well. Am I missing something?

Reply:
@srhb Both `with-packages` and all Idris packages are in `pkgs.idrisPackages`, this would turn out like this:

```
nix-env -iE 'pkgs: pkgs.idrisPackages.with-packages (with pkgs.idrisPackages; [ contrib pruviloj ])'
```

I'd personally prefer the current, less verbose option, but I do see the argument being made. I'll probably change this later